extensions/applicationinsights-cfgsync-js/src/CfgSyncPlugin.ts

```diff
@@ -0,0 +1,412 @@
+                _eventOff();
+                _clearScheduledTimer();
+                _initDefaults();
+            };
+
+            _self["_getDbgPlgTargets"] = () => {
+                return [_broadcastChanges, _receiveChanges, _evtName];
+            };
+    
+            function _initDefaults() {
+                _mainConfig = null;
+                _evtName = null;
+                _evtNamespace = null;
+                _cfgUrl = null;
+                _receiveChanges = null;
+                _broadcastChanges = null;
+                _nonOverrideConfigs = null;
+                _timeoutHandle = null;
+                _fetchSpan = null;
+                _retryCnt = null;
+                _overrideFetchFn = null;
+                _overrideSyncFn = null;
+                _onCfgChangeReceive = null;
+            }
+
+            function _populateDefaults(config: IConfiguration) {
+                let identifier = _self.identifier;
+                let core = _self.core;
+               
+                _self._addHook(onConfigChange(config, () => {
+                    let ctx = createProcessTelemetryContext(null, config, core);
+                    _extensionConfig = ctx.getExtCfg(identifier, _defaultConfig);
+
+                    if (isNullOrUndefined(_receiveChanges)) {
+                        _receiveChanges = _extensionConfig.syncMode === ICfgSyncMode.Receive;
+                    }
+                    if (isNullOrUndefined(_broadcastChanges)) {
+                        _broadcastChanges = _extensionConfig.syncMode === ICfgSyncMode.Broadcast;
+                    }
+                   
+                    let _newEvtName =  _extensionConfig.customEvtName || evtName;
+                    if (_evtName !== _newEvtName) {
+                        if (_receiveChanges) {
+                            _updateEventListenerName(_newEvtName);
+
+                        } else {
+                            _eventOff();
+                            _evtName = _newEvtName;
+                        }
+
+                    }
+
+                    if (isNullOrUndefined(_cfgUrl)) {
+                        _cfgUrl = _extensionConfig.cfgUrl;
+                    }
+                    // if cfgUrl is set, we will ignore core config change
+                    if (!_cfgUrl) {
+                        _mainConfig = config;
+                        
+                        if (_broadcastChanges) {
+                            objExtend({}, config);
+                            _sendCfgsyncEvents();
+                        }
+                    }
+                }));
+               
+                _overrideSyncFn = _extensionConfig.overrideSyncFn;
+                _overrideFetchFn = _extensionConfig.overrideFetchFn;
+                _onCfgChangeReceive = _extensionConfig.onCfgChangeReceive;
+                _nonOverrideConfigs = _extensionConfig.nonOverrideConfigs;
+                _fetchSpan = _extensionConfig.scheduleFetchTimeout;
```

Comment:
nit: lets call this _fetchTimeout or _fetchFrequency as span is going to have a different meaning in the future once we start supporting OpenTelementry "Spans"